File: src/test/java/org/hisp/dhis/rules/models/RuleVariableCurrentEventTests.java
package org.hisp.dhis.rules.models;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.JUnit4;

import static org.assertj.core.api.Java6Assertions.assertThat;

@RunWith( JUnit4.class )
public class RuleVariableCurrentEventTests
{

        @Test(expected = NullPointerException.class )
        public void createShouldThrowOnNullName()
        {
                RuleVariableCurrentEvent.create( null, "test_dataelement", RuleValueType.TEXT );
        }

        @Test(expected = NullPointerException.class )
        public void createShouldThrowOnNullDataElement()
        {
                RuleVariableCurrentEvent.create( "test_variable", null, RuleValueType.TEXT );
        }

        @Test(expected = NullPointerException.class )
        public void createShouldThrowOnNullDataElementType()
        {
                RuleVariableCurrentEvent.create( "test_variable", "test_dataelement", null );
        }

        @Test
        public void createShouldPropagatePropertiesCorrectly()
        {
                RuleVariableCurrentEvent ruleVariableCurrentEvent = RuleVariableCurrentEvent.create(
                    "test_variable", "test_dataelement", RuleValueType.NUMERIC );

                assertThat( ruleVariableCurrentEvent.name() ).isEqualTo( "test_variable" );
                assertThat( ruleVariableCurrentEvent.dataElement() ).isEqualTo( "test_dataelement" );
                assertThat( ruleVariableCurrentEvent.dataElementType() ).isEqualTo( RuleValueType.NUMERIC );
        }
}

File: src/jvmTest/kotlin/org/hisp/dhis/rules/models/RuleVariableCurrentEventTests.kt
package org.hisp.dhis.rules.models

import org.assertj.core.api.Java6Assertions.assertThat
import org.junit.Test
import org.junit.runner.RunWith
import org.junit.runners.JUnit4
import kotlin.test.assertFailsWith

@RunWith(JUnit4::class)
class RuleVariableCurrentEventTests {

    @Test
    fun createShouldThrowOnNullName() {
        assertFailsWith<NullPointerException> {
            RuleVariableCurrentEvent.create(null!!, "test_dataelement", RuleValueType.TEXT)
        }
    }

    @Test
    fun createShouldThrowOnNullDataElement() {
        assertFailsWith<NullPointerException> {
            RuleVariableCurrentEvent.create("test_variable", null!!, RuleValueType.TEXT)
        }
    }

    @Test
    fun createShouldThrowOnNullDataElementType() {
        assertFailsWith<NullPointerException> {
            RuleVariableCurrentEvent.create("test_variable", "test_dataelement", null!!)
        }
    }

    @Test
    fun createShouldPropagatePropertiesCorrectly() {
        val ruleVariableCurrentEvent = RuleVariableCurrentEvent.create(
                "test_variable", "test_dataelement", RuleValueType.NUMERIC)

        assertThat(ruleVariableCurrentEvent.name).isEqualTo("test_variable")
        assertThat(ruleVariableCurrentEvent.dataElement).isEqualTo("test_dataelement")
        assertThat(ruleVariableCurrentEvent.dataElementType).isEqualTo(RuleValueType.NUMERIC)
    }
}
